src/Workspaces/Remote/Core/Services/AssetService.cs

```diff
@@ -17,39 +17,70 @@ namespace Microsoft.CodeAnalysis.Remote
     /// </summary>
     internal class AssetService
     {
+        private const int CleanupInterval = 30000; // 30 seconds
```

Comment:
I think this is too short considering running analyzers on a compilation avg takes about 18 seconds. this will make us to let goes a lot of data all the time.